### PR TITLE
ADZ-1190 API-spec side nav rebrand

### PR DESCRIPTION
**Pull request by urchy**
Screenshot 2021-06-21 at 11 36 21

### CODE REVIEW

**Review by ziemsky — APPROVED** [review]
Side-nav looks great!

File: cms/src/test/resources/test-data/api-specifications/SwaggerCodeGenOpenApiSpecificationJsonToHtmlConverterTest/oasV3_operationIdsInVariousFormats.html [hunk 1]
```diff
@@ -1,77 +1,102 @@
...
-            </div>
-        </div>
+    <div class="nhsd-m-sticky-navigation nhsd-!t-display-sticky nhsd-!t-display-sticky--offset-2 nhsd-!t-margin-bottom-6">
+        <span id="sticky-nav-header" class="nhsd-t-heading-xs nhsd-!t-margin-bottom-2">Page content</span>
+        <hr class="nhsd-a-horizontal-rule nhsd-a-horizontal-rule--size-xs nhsd-!t-margin-bottom-2"/>
+        <nav aria-labelledby="sticky-nav-header">
+            <ul>
+                <li class="nhsd-m-sticky-navigation__item" data-nav-content="Top of page">
+                    <a class="nhsd-a-link" href="#top">Top of page
+                        <span class="nhsd-t-sr-only"></span>
```
Comment (LEJA3) [inline cms/src/test/resources/test-data/api-specifications/SwaggerCodeGenOpenApiSpecificationJsonToHtmlConverterTest/oasV3_operationIdsInVariousFormats.html:10]: Element not needed

File: cms/src/test/resources/test-data/api-specifications/SwaggerCodeGenOpenApiSpecificationJsonToHtmlConverterTest/oasV3_operationIdsInVariousFormats.html [hunk 2]
```diff
@@ -1,77 +1,102 @@
...
+                    </a>
+                </li>
+                <li class="nhsd-m-sticky-navigation__item nhsd-m-sticky-navigation__item--nested" data-nav-content="api-Patients-camelCase">
+                    <a class="nhsd-a-link"
+                       href="#api-Patients-camelCase"
+                       aria-label="camelCase"
+                       title="camelCase">Get camelCase operation
+                        <span class="nhsd-t-sr-only"></span>
+                    </a>
+                </li>
```
Comment (LEJA3) [inline cms/src/test/resources/test-data/api-specifications/SwaggerCodeGenOpenApiSpecificationJsonToHtmlConverterTest/oasV3_operationIdsInVariousFormats.html:92]: Same link accessibility issues as above.

File: cms/src/test/resources/test-data/api-specifications/SwaggerCodeGenOpenApiSpecificationJsonToHtmlConverterTest/oasV3_operationsNotTagged.html [hunk 3]
```diff
@@ -1,52 +1,62 @@
...
-            </div>
-        </div>
+    <div class="nhsd-m-sticky-navigation nhsd-!t-display-sticky nhsd-!t-display-sticky--offset-2 nhsd-!t-margin-bottom-6">
+        <span id="sticky-nav-header" class="nhsd-t-heading-xs nhsd-!t-margin-bottom-2">Page content</span>
+        <hr class="nhsd-a-horizontal-rule nhsd-a-horizontal-rule--size-xs nhsd-!t-margin-bottom-2"/>
+        <nav aria-labelledby="sticky-nav-header">
+            <ul>
+                <li class="nhsd-m-sticky-navigation__item" data-nav-content="Top of page">
+                    <a class="nhsd-a-link" href="#top">Top of page
+                        <span class="nhsd-t-sr-only"></span>
```
Comment (LEJA3) [inline cms/src/test/resources/test-data/api-specifications/SwaggerCodeGenOpenApiSpecificationJsonToHtmlConverterTest/oasV3_operationsNotTagged.html:10]: Element not needed

File: cms/src/test/resources/test-data/api-specifications/SwaggerCodeGenOpenApiSpecificationJsonToHtmlConverterTest/oasV3_operationsNotTagged.html [hunk 4]
```diff
@@ -1,52 +1,62 @@
...
+                    </a>
+                </li>
+                <li class="nhsd-m-sticky-navigation__item nhsd-m-sticky-navigation__item--nested" data-nav-content="api-Default-path_bPost">
+                    <a class="nhsd-a-link"
+                       href="#api-Default-path_bPost"
+                       aria-label="path_bPost"
+                       title="path_bPost">Post operation B
+                        <span class="nhsd-t-sr-only"></span>
+                    </a>
+                </li>
```
Comment (LEJA3) [inline cms/src/test/resources/test-data/api-specifications/SwaggerCodeGenOpenApiSpecificationJsonToHtmlConverterTest/oasV3_operationsNotTagged.html:52]: Same link accessibility issues as above.

File: cms/src/test/resources/test-data/api-specifications/SwaggerCodeGenOpenApiSpecificationJsonToHtmlConverterTest/oasV3_operationsTagged.html [hunk 5]
```diff
@@ -1,62 +1,78 @@
...
-            </div>
-        </div>
+    <div class="nhsd-m-sticky-navigation nhsd-!t-display-sticky nhsd-!t-display-sticky--offset-2 nhsd-!t-margin-bottom-6">
+        <span id="sticky-nav-header" class="nhsd-t-heading-xs nhsd-!t-margin-bottom-2">Page content</span>
+        <hr class="nhsd-a-horizontal-rule nhsd-a-horizontal-rule--size-xs nhsd-!t-margin-bottom-2"/>
+        <nav aria-labelledby="sticky-nav-header">
+            <ul>
+                <li class="nhsd-m-sticky-navigation__item" data-nav-content="Top of page">
+                    <a class="nhsd-a-link" href="#top">Top of page
+                        <span class="nhsd-t-sr-only"></span>
```
Comment (LEJA3) [inline cms/src/test/resources/test-data/api-specifications/SwaggerCodeGenOpenApiSpecificationJsonToHtmlConverterTest/oasV3_operationsTagged.html:10]: Element not needed

File: cms/src/test/resources/test-data/api-specifications/SwaggerCodeGenOpenApiSpecificationJsonToHtmlConverterTest/oasV3_operationsTagged.html [hunk 6]
```diff
@@ -1,62 +1,78 @@
...
+                    </a>
+                </li>
+                <li class="nhsd-m-sticky-navigation__item nhsd-m-sticky-navigation__item--nested" data-nav-content="api-ResourceB-path_bGet">
+                    <a class="nhsd-a-link"
+                       href="#api-ResourceB-path_bGet"
+                       aria-label="path_bGet"
+                       title="path_bGet">Get operation B
+                        <span class="nhsd-t-sr-only"></span>
+                    </a>
+                </li>
```
Comment (LEJA3) [inline cms/src/test/resources/test-data/api-specifications/SwaggerCodeGenOpenApiSpecificationJsonToHtmlConverterTest/oasV3_operationsTagged.html:68]: Same link accessibility issues as above

File: cms/src/test/resources/test-data/api-specifications/SwaggerCodeGenOpenApiSpecificationJsonToHtmlConverterTest/oasV3_withNoRequest.html [hunk 7]
```diff
@@ -1,37 +1,38 @@
...
-            </div>
-        </div>
+    <div class="nhsd-m-sticky-navigation nhsd-!t-display-sticky nhsd-!t-display-sticky--offset-2 nhsd-!t-margin-bottom-6">
+        <span id="sticky-nav-header" class="nhsd-t-heading-xs nhsd-!t-margin-bottom-2">Page content</span>
+        <hr class="nhsd-a-horizontal-rule nhsd-a-horizontal-rule--size-xs nhsd-!t-margin-bottom-2"/>
+        <nav aria-labelledby="sticky-nav-header">
+            <ul>
+                <li class="nhsd-m-sticky-navigation__item" data-nav-content="Top of page">
+                    <a class="nhsd-a-link" href="#top">Top of page
+                        <span class="nhsd-t-sr-only"></span>
```
Comment (LEJA3) [inline cms/src/test/resources/test-data/api-specifications/SwaggerCodeGenOpenApiSpecificationJsonToHtmlConverterTest/oasV3_withNoRequest.html:10]: Element not needed

File: cms/src/test/resources/test-data/api-specifications/SwaggerCodeGenOpenApiSpecificationJsonToHtmlConverterTest/oasV3_withNoRequest.html [hunk 8]
```diff
@@ -1,37 +1,38 @@
...
+                    </a>
+                </li>
+                <li class="nhsd-m-sticky-navigation__item nhsd-m-sticky-navigation__item--nested" data-nav-content="api-Default-search-patient">
+                    <a class="nhsd-a-link"
+                       href="#api-Default-search-patient"
+                       aria-label="search-patient"
+                       title="search-patient">Search for patient
+                        <span class="nhsd-t-sr-only"></span>
+                    </a>
+                </li>
```
Comment (LEJA3) [inline cms/src/test/resources/test-data/api-specifications/SwaggerCodeGenOpenApiSpecificationJsonToHtmlConverterTest/oasV3_withNoRequest.html:28]: Same link accessibility issues as above.

File: cms/src/test/resources/test-data/api-specifications/SwaggerCodeGenOpenApiSpecificationJsonToHtmlConverterTest/oasV3_withNoResponse.html [hunk 9]
```diff
@@ -1,36 +1,38 @@
...
-            </div>
-        </div>
+    <div class="nhsd-m-sticky-navigation nhsd-!t-display-sticky nhsd-!t-display-sticky--offset-2 nhsd-!t-margin-bottom-6">
+        <span id="sticky-nav-header" class="nhsd-t-heading-xs nhsd-!t-margin-bottom-2">Page content</span>
+        <hr class="nhsd-a-horizontal-rule nhsd-a-horizontal-rule--size-xs nhsd-!t-margin-bottom-2"/>
+        <nav aria-labelledby="sticky-nav-header">
+            <ul>
+                <li class="nhsd-m-sticky-navigation__item" data-nav-content="Top of page">
+                    <a class="nhsd-a-link" href="#top">Top of page
+                        <span class="nhsd-t-sr-only"></span>
```
Comment (LEJA3) [inline cms/src/test/resources/test-data/api-specifications/SwaggerCodeGenOpenApiSpecificationJsonToHtmlConverterTest/oasV3_withNoResponse.html:10]: Element not needed.

File: cms/src/test/resources/test-data/api-specifications/SwaggerCodeGenOpenApiSpecificationJsonToHtmlConverterTest/oasV3_withNoResponse.html [hunk 10]
```diff
@@ -1,36 +1,38 @@
...
+                    </a>
+                </li>
+                <li class="nhsd-m-sticky-navigation__item nhsd-m-sticky-navigation__item--nested" data-nav-content="api-Default-search-patient">
+                    <a class="nhsd-a-link"
+                       href="#api-Default-search-patient"
+                       aria-label="search-patient"
+                       title="search-patient">Search for patient
+                        <span class="nhsd-t-sr-only"></span>
+                    </a>
+                </li>
```
Comment (LEJA3) [inline cms/src/test/resources/test-data/api-specifications/SwaggerCodeGenOpenApiSpecificationJsonToHtmlConverterTest/oasV3_withNoResponse.html:28]: Same link accessibility issues as above.

**Review by LEJA3 — COMMENTED** [review]
The code looks good, but I noticed a few minor accessibility issues, mostly on sticky nav links.

Also, I'm aware the toolkit is currently showing an empty `nhsd-t-sr-only` element in the example HTML for links that don't require it so apologies for that. I'll push up a change to remove that from the toolkit today.

**Review by LEJA3 — APPROVED** [review]
Thanks for updating. Looks good.

**Pull request merged**